### PR TITLE
Add verification for accessing chat, delete account feature and header image for sport activity

### LINKED ISSUE #137: Add organizer full name in activity description

### LINKED ISSUE #145: Allow only registered users to access chat

### LINKED ISSUE #147: Delete user account functionality

### LINKED ISSUE #150: Add possibility to have a header image for the sport activities once it has been created (this capture will later be locally cached)

### PR DESCRIPTION
**Pull request by Antho1426**
Work in progress...

The verification for accessing the chat works (we have to be registered to access the chat), but currently, the registered users are not uploaded to Firebase Firestore. More precisely, the "list field" participantId doesn't update when we register using the "REGISTER" button in ActivityDescriptionActivity... We will soon have to check how we can update the fields of an activity in Firebase (to really save the number of participants and their userId).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,8 +2,15 @@
...
 ### SDP - spring semester 2021
 
+## Build
 [![Build Status](https://api.cirrus-ci.com/github/movemeet/movemeet.svg)](https://cirrus-ci.com/github/movemeet/movemeet)
 
+## Maintainability
+[![Maintainability](https://api.codeclimate.com/v1/badges/23a7b536f058d3bfda16/maintainability)](https://codeclimate.com/github/movemeet/movemeet/maintainability)
+
+## Test Coverage
+[![Test Coverage](https://api.codeclimate.com/v1/badges/23a7b536f058d3bfda16/test_coverage)](https://codeclimate.com/github/movemeet/movemeet/test_coverage)
```
Comment (RoxaneBurri) [inline README.md:12]: Nice ! We can see our coverge !

File: app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java [hunk 2]
```diff
@@ -273,13 +290,49 @@ public void onClick(View v) {
...
 
     public void goToIndividualChat(View view) {
-        Intent intent = new Intent(ActivityDescriptionActivity.this, ChatActivity.class);
-        String activityChatId = act.getActivityId() + " - chatId";
-        intent.putExtra("ACTIVITY_CHAT_ID", activityChatId);
-        String activityTitle = act.getTitle();
-        intent.putExtra("ACTIVITY_TITLE", activityTitle);
-        startActivity(intent);
+        // Allowing the access to the chat only if the user is registered to the activity
+        if (act.getParticipantId().contains(userId)) {
```
Comment (RoxaneBurri) [inline app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java:289]: Good check, we need it

File: app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java [hunk 3]
```diff
@@ -273,13 +290,49 @@ public void onClick(View v) {
...
+    }
+
+    private void loadActivityHeaderPicture() {
+        activityImage = findViewById(R.id.activity_image_description);
+        progressBar = findViewById(R.id.progress_bar_activity_description);
+        progressBar.setVisibility(View.VISIBLE);
+        storageReference = FirebaseStorage.getInstance().getReference();
+        if (act != null) {
+            imagePath = "activities/"+act.getActivityId()+"/activityImage.jpg";
+            StorageReference imageRef = storageReference.child(imagePath);
```
Comment (RoxaneBurri) [inline app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java:337]: You don't store the image in the activity ?

File: app/src/main/java/com/sdp/movemeet/Backend/FirebaseInteraction.java [hunk 4]
```diff
@@ -87,4 +90,44 @@ public static void logoutIfUserNonNull(FirebaseAuth fAuth, Activity activity) {
         }
     }
 
+    public static void getImageFromFirebase(StorageReference imageRef, ImageView imageView, ProgressBar progressBar) {
```
Comment (RoxaneBurri) [inline app/src/main/java/com/sdp/movemeet/Backend/FirebaseInteraction.java:93]: This is an interface for get and upload an image in Firebase ?

File: app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java (hunk 2 above)
Comment (Antho1426) [reply app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java:289]: In fact, this is key for the chat of the sport activity. We don't want intruders to join! 😝

File: app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java (hunk 3 above)
Comment (Antho1426) [reply app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java:337]: Right for the moment the image is solely stored on Firebase Storage. It is anyway linked (referenced with a unique path containing the sport activity ID) to the sport activity as its "header picture", but for the moment we don't have a local cache caching the "header pictures" of the sport activities! This is the next step.

File: app/src/main/java/com/sdp/movemeet/Backend/FirebaseInteraction.java (hunk 4 above)
Comment (Antho1426) [reply app/src/main/java/com/sdp/movemeet/Backend/FirebaseInteraction.java:93]: Well, this is simply a refactored method implemented in the FirebaseInteraction class to avoid code duplicates (because this "getImageFromFirebase" function is also used in EditProfileActivity).

File: README.md [hunk 5]
```diff
@@ -2,8 +2,15 @@
 
 ### SDP - spring semester 2021
 
+## Build
 [![Build Status](https://api.cirrus-ci.com/github/movemeet/movemeet.svg)](https://cirrus-ci.com/github/movemeet/movemeet)
 
+## Maintainability
+[![Maintainability](https://api.codeclimate.com/v1/badges/23a7b536f058d3bfda16/maintainability)](https://codeclimate.com/github/movemeet/movemeet/maintainability)
+
```
Comment (kotleta2007) [inline README.md:10]: It will look marvelous after it's merged.

File: app/src/main/java/com/sdp/movemeet/EditProfileActivity.java [hunk 6]
```diff
@@ -88,14 +88,9 @@ private void assignViewsAndAdjustData() {
 
     private void loadRegisteredUserProfilePicture(String userId) {
         progressBar.setVisibility(View.VISIBLE);
-        StorageReference profileRef = storageReference.child("users/"+userId+"/profile.jpg");
-        profileRef.getDownloadUrl().addOnSuccessListener(new OnSuccessListener<Uri>() {
-            @Override
-            public void onSuccess(Uri uri) {
```
Comment (kotleta2007) [inline app/src/main/java/com/sdp/movemeet/EditProfileActivity.java:94]: Why did you remove this part? Was it refactored into FirebaseInteraction?

**Review by kotleta2007 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (jlstoupy98) [inline README.md:12]: At least we will have a nice readme xD

**Review by RoxaneBurri — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (Antho1426) [reply README.md:10]: Indeed, that's the easy part ^^

File: app/src/main/java/com/sdp/movemeet/EditProfileActivity.java (hunk 6 above)
Comment (Antho1426) [reply app/src/main/java/com/sdp/movemeet/EditProfileActivity.java:94]: That's exactly this! This is because I use the same method in ActivityDescriptionActivity and and EditProfileActivity to pick a picture from the Library.

File: README.md (hunk 1 above)
Comment (Antho1426) [reply README.md:12]: Alright!! ^^👍

File: app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java [hunk 7]
```diff
@@ -272,13 +284,81 @@ public void onClick(View v) {
...
+            intent.putExtra("ACTIVITY_TITLE", activityTitle);
+            startActivity(intent);
+        } else {
+            Toast.makeText(ActivityDescriptionActivity.this, "Please register if you want to access the chat!", Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    private void getOrganizerName() {
+        if (act != null) {
+            organizerId = act.getOrganizerId();
```
Comment (Walfar) [inline app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java:303]: What is the use of getting the organizer Id if the view is his username ?

Comment (Antho1426) [reply app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java:303]: Because the organiser ID (as well as for instance the user ID) is the key to access the organiser (respectively the user) data on Firebase Firestore. More precisely, the Firebase Firestore document is referenced through the organiser (respectively the user) ID:
DocumentReference docRef = fStore.collection("users").document(organizerId);

**Review by Walfar — COMMENTED** [review]
Looks all good to me :) although we'll have to look how to refactor all that with the Firebase backend interface
